frontend/src/views/setting/panel/index.vue

```diff
@@ -440,36 +440,36 @@ const onChangeThemeColor = () => {
 
 const onSave = async (key: string, val: any) => {
     loading.value = true;
-    if (key === 'Language') {
-        i18n.locale.value = val;
-        globalStore.updateLanguage(val);
-    }
-    if (key === 'Theme') {
-        globalStore.themeConfig.theme = val;
-        switchTheme();
-        if (globalStore.isProductPro) {
-            await updateXpackSettingByKey('Theme', val);
-            let color: string;
-            const themeColor: ThemeColor = JSON.parse(globalStore.themeConfig.themeColor);
-            if (val === 'auto') {
-                const prefersDark = window.matchMedia('(prefers-color-scheme: dark)');
-                color = prefersDark.matches ? themeColor.dark : themeColor.light;
-            } else {
-                color = val === 'dark' ? themeColor.dark : themeColor.light;
-            }
-            globalStore.themeConfig.primary = color;
-            setPrimaryColor(color);
-        }
-    }
-    if (key === 'MenuTabs') {
-        globalStore.setOpenMenuTabs(val === 'enable');
-    }
     let param = {
         key: key,
         value: val + '',
     };
     await updateSetting(param)
         .then(async () => {
+            if (key === 'Language') {
+                i18n.locale.value = val;
+                globalStore.updateLanguage(val);
+            }
+            if (key === 'Theme') {
+                globalStore.themeConfig.theme = val;
+                switchTheme();
+                if (globalStore.isProductPro) {
+                    await updateXpackSettingByKey('Theme', val);
+                    let color: string;
+                    const themeColor: ThemeColor = JSON.parse(globalStore.themeConfig.themeColor);
+                    if (val === 'auto') {
+                        const prefersDark = window.matchMedia('(prefers-color-scheme: dark)');
+                        color = prefersDark.matches ? themeColor.dark : themeColor.light;
+                    } else {
+                        color = val === 'dark' ? themeColor.dark : themeColor.light;
+                    }
+                    globalStore.themeConfig.primary = color;
+                    setPrimaryColor(color);
+                }
+            }
+            if (key === 'MenuTabs') {
+                globalStore.setOpenMenuTabs(val === 'enable');
+            }
             if (param.key === 'Language') {
                 location.reload();
             }
```

Comment:
The provided code has no major irregularities or potential issues that I can identify based on the given changes. The differences you've made are mostly identical, which means nothing is changing beyond what was already there. There's no need for optimization suggestions in this case, as the functions and conditional logic remain unchanged.

Here’s the relevant part again for clarity:

```javascript
const onChangeThemeColor = () => {
}

const onSave = async (key: string, val: any) => {
    loading.value = true;

    let param = {
        key: key,
        value: val + '',
    };
    await updateSetting(param)
        .then(async () => {
            if (key === 'Language') {
                i18n.locale.value = val;
                globalStore.updateLanguage(val);
            }
            if (key === 'Theme') {
                globalStore.themeConfig.theme = val;
                switchTheme();
                if (globalStore.isProductPro) {
                    await updateXpackSettingByKey('Theme', val);
                    let color: string;
                    const themeColor: ThemeColor = JSON.parse(globalStore.themeConfig.themeColor);
                    if (val === 'auto') {
                        const prefersDark = window.matchMedia('(prefers-color-scheme: dark)');
                        color = prefersDark.matches ? themeColor.dark : themeColor.light;
                    } else {
                        color = val === 'dark' ? themeColor.dark : themeColor.light;
                    }
                    globalStore.themeConfig.primary = color;
                    setPrimaryColor(color);
                }
            }
            if (key === 'MenuTabs') {
                globalStore.setOpenMenuTabs(val === 'enable');
            }

            if (param.key === 'Language') {
                location.reload();
            }
        })
        .finally(() => {
            loading.value = false;
        });
}
```

If this function were to be modified in the future, it might make sense to refactor some of these repeated blocks into helper functions or use more efficient data structures where appropriate. However, without additional context about the larger application architecture or specific requirements, those would not seem necessary at this point.